apollo-router/src/query_planner/plan.rs

```diff
@@ -186,6 +186,40 @@ impl PlanNode {
         }
     }
 
+    pub(crate) fn subgraph_fetches(&self) -> usize {
+        match self {
+            PlanNode::Sequence { nodes } => nodes.iter().map(|n| n.subgraph_fetches()).sum(),
+            PlanNode::Parallel { nodes } => nodes.iter().map(|n| n.subgraph_fetches()).sum(),
+            PlanNode::Fetch(_) => 1,
+            PlanNode::Flatten(node) => node.node.subgraph_fetches(),
+            PlanNode::Defer { primary, deferred } => {
+                primary.node.as_ref().map_or(0, |n| n.subgraph_fetches())
+                    + deferred
+                        .iter()
+                        .map(|n| n.node.as_ref().map_or(0, |n| n.subgraph_fetches()))
+                        .sum::<usize>()
+            }
+            PlanNode::Subscription { rest, .. } => {
+                rest.as_ref().map_or(0, |n| n.subgraph_fetches())
```

Comment:
The primary should be counted as a subgraph fetch I think. It's doing a request to a subgraph

Reply:
I've added a `+ 1` and small comment explaining this